sql/core/src/test/scala/org/apache/spark/sql/DataFrameSuite.scala

```diff
@@ -684,6 +684,34 @@ class DataFrameSuite extends QueryTest with SharedSQLContext {
     }
   }
 
```

Comment:
I noticed we don't have a test for the single column `withColumn: given metadata`. I wonder if that should be added (though it's not related to this PR)

Reply:
I think as `withColumn` case, we can re-implement it with `withColumns` for metadata too. So this test case can cover it.